src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -473,13 +473,21 @@ public void send(PacketInfo packetInfo)
             // we can't just reassign a transformed packet back into its
             // proper PacketInfo. Need to change those classes to work with
             // the new packet types
-            VideoRtpPacket[] res = bitrateController.transformRtp(packetInfo);
-            for (VideoRtpPacket videoRtpPacket : res)
+            VideoRtpPacket[] extras = bitrateController.transformRtp(packetInfo);
+            if (extras == null)
+            {
+                logger.warn(
+                    "Dropping a packet which was supposed to be accepted:"
+                        + packet);
+                return;
+            }
+
+            // The original packet was transformed in place.
+            // TODO: should we send this *after* the extras?
+            transceiver.sendPacket(packetInfo);
```

Comment:
wait, does the returned array not already include the original packet?  given that the 'extras' may contain packets _before_ this packetInfo, this feels weird to me.  the bitratecontroller is the best to determine the order in which the packets it returns should be sent (since it's aware of things being piggybacked/older packets now being forwarded) so it seems like we should trust its return value rather than sending the original in place and then the extras

Reply:
(ok, i see you changed the behavior below about not including the original packet, but i still don't get the point of this change...does it bring us closer to how we want things to end up?)

Reply:
"does it bring us closer to how we want things to end up?" 
Yes, the PacketInfo instance is preserved, except for piggybacked packets. So preserved for the vast majority of packets.